### ISSUE TITLE
Default charset might break scalatex

### ISSUE THREAD

**Issue by romainreuillon** [lihaoyi/Scalatex#10]
When building on machin with high level of concurency I often get this kind of error, despite the build works perfecty on more sequential computers:

```
[error]   def apply(): Frag = scalatex.twf("/tmp/openmole/openmole/bin/org.openmole.site/src/main/scalatex/org.openmole.site/documentation/console/tutorial/HeadlessNetLogo.scalatex")
[error]                                   ^
[error] /tmp/openmole/openmole/bin/org.openmole.site/target/scala-2.11/src_managed/main/scalatex/org.openmole.site/documentation/console/tutorial/HelloWorld.scala:9: exception during macro expansion: 
[error] java.nio.charset.MalformedInputException: Input length = 1
[error] 	at java.nio.charset.CoderResult.throwException(CoderResult.java:281)
[error] 	at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:339)
[error] 	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
[error] 	at java.io.InputStreamReader.read(InputStreamReader.java:184)
[error] 	at java.io.BufferedReader.read1(BufferedReader.java:210)
[error] 	at java.io.BufferedReader.read(BufferedReader.java:286)
[error] 	at java.io.Reader.read(Reader.java:140)
[error] 	at scala.io.BufferedSource.mkString(BufferedSource.scala:96)
[error] 	at scalatex.package$Internals$.applyMacroFileBase(package.scala:36)
[error] 	at scalatex.package$Internals$.applyMacroFile(package.scala:25)
[error] 	at sun.reflect.GeneratedMethodAccessor14.invoke(Unknown Source)
[error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.lang.reflect.Method.invoke(Method.java:497)
[error] 	at scala.reflect.macros.runtime.JavaReflectionRuntimes$JavaReflectionResolvers$$anonfun$resolveJavaReflectionRuntime$2.apply(JavaReflectionRuntimes.scala:34)
[error] 	at scala.reflect.macros.runtime.JavaReflectionRuntimes$JavaReflectionResolvers$$anonfun$resolveJavaReflectionRuntime$2.apply(JavaReflectionRuntimes.scala:22)
[error] 	at scala.tools.nsc.typechecker.Macros$class.macroExpandWithRuntime(Macros.scala:755)
```

**Comment by romainreuillon** [comment]
... in fact may be it comes from the local settings of the server and not the parallel aspect.

**Comment by romainreuillon** [comment]
It was a charset issue, adding -Dfile.encoding=UTF-8 fixed the pb.

**Comment by lihaoyi** [comment]
Ok cool. Any idea what Scalatex can do to help avoid this issue? All this encoding stuff is black magic to me -.-

**Comment by romainreuillon** [comment]
Yes, sources gets an implict Codec when they are consturcted, you can provide an UTF-8 codec instead of the platform default one. I guess you should do the same when writting files.

**Comment by lihaoyi** [comment]
Cool, want to send a PR? I could blind-commit a fix but I won't actually know if I actually fixed anything...

**Comment by lihaoyi** [comment]
Fixed in 0.3.1 AFAIK
